src/fiskaltrust.Middleware.Demo/RestPos.cs

```diff
@@ -0,0 +1,210 @@
+            _url = url.StartsWith("rest://") ? url.Replace("rest://", "http://") : url.Replace("xml://", "http://");
+            _requestType = url;
+        }
+
+        IAsyncResult ifPOS.v0.IPOS.BeginEcho(string message, AsyncCallback callback, object state)
+        {
+            throw new NotSupportedException();
+        }
+
+        IAsyncResult ifPOS.v0.IPOS.BeginJournal(long ftJournalType, long from, long to, AsyncCallback callback, object state)
+        {
+            throw new NotSupportedException();
+        }
+
+        IAsyncResult ifPOS.v0.IPOS.BeginSign(ifPOS.v0.ReceiptRequest data, AsyncCallback callback, object state)
+        {
+            throw new NotSupportedException();
+        }
+
+        string ifPOS.v0.IPOS.Echo(string message)
+        {       
+
+            using (var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri(_url);
+                
+                var data = new { message = message};
+                var jsonstring = JsonConvert.SerializeObject(data);
+                var jsonContent = new StringContent(jsonstring, Encoding.UTF8, "application/json");                                            
+
+                using (var response = client.PostAsync("v0/Echo", jsonContent).Result)
+                {
+                    var reponse = response.Content.ReadAsStringAsync();
+                    return reponse.ToString();
+                }
+            }
+        }       
+
+        async Task<EchoResponse> ifPOS.v1.IPOS.EchoAsync(EchoRequest message)
+        {
+            if (_requestType.StartsWith("rest://"))
+            {
+                return await JsonEchoAsync(message);
+            }
+            else
+            {
+                return await XmlEchoAsync(message);
+            }
+
+        }
+
+        private async Task<EchoResponse> XmlEchoAsync(EchoRequest message)
+        {
+            var xmlString = Serialize(message);
+            var xmlContent = new StringContent(xmlString, Encoding.UTF8, "application/xml");
+
+            using (var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri(_url);
+
+                using (var response = await client.PostAsync("v1/Echo", xmlContent))
+                {
+                    var content = await response.Content.ReadAsStringAsync();
+
+                    var xml = XElement.Parse(content);
+                    string jsonText = JsonConvert.SerializeXNode(xml);
+                    return JsonConvert.DeserializeObject<EchoResponse>(jsonText);
+                }
+            }
+        }
+
+        private async Task<EchoResponse> JsonEchoAsync(EchoRequest message)
+        {
+            var jsonstring = JsonConvert.SerializeObject(message);
+            var jsonContent = new StringContent(jsonstring, Encoding.UTF8, "application/json");
+
+            using (var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri(_url);
```

Comment:
Since we do have urls with a path most of the time (e.g `http://localhost:1500/4417bfc4-cb55-4813-8914-1665cff5187e`) and the BaseAddress strips of the last part, I think we should use `Path.Combine(_url, "v1/echo")` instead.